In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [3]:
education_districtwise = pd.read_csv("education_districtwise.csv")
education_districtwise = education_districtwise.dropna()

In [6]:
education_districtwise.head(5)

DISTNAME STATNAME  BLOCKS  VILLAGES  CLUSTERS  TOTPOPULAT  OVERALL_LI
0   DISTRICT32   STATE1      13       391       104    875564.0       66.92
1  DISTRICT649   STATE1      18       678       144   1015503.0       66.93
2  DISTRICT229   STATE1       8        94        65   1269751.0       71.21
3  DISTRICT259   STATE1      13       523       104    735753.0       57.98
4  DISTRICT486   STATE1       8       359        64    570060.0       65.00

In [10]:
sampled_data = education_districtwise.sample(n = 50, replace = True, random_state = 31208)
sampled_data

DISTNAME STATNAME  BLOCKS  VILLAGES  CLUSTERS  TOTPOPULAT  OVERALL_LI
661  DISTRICT528   STATE6       9       112        89   1863174.0       92.14
216  DISTRICT291  STATE28      14      1188       165   3273127.0       52.49
367   DISTRICT66  STATE23      12      1169       116   1042304.0       62.14
254  DISTRICT458   STATE3       3       157        19     82839.0       76.33
286  DISTRICT636  STATE35       3       187        44    514683.0       86.70
369  DISTRICT512  STATE23       6       589        30    717169.0       68.35
258  DISTRICT156   STATE3       6        80         9     35289.0       59.94
10   DISTRICT412   STATE1      11       187        95    476820.0       68.69
512  DISTRICT277   STATE9      10       558       179   2298934.0       84.31
144  DISTRICT133  STATE21      14      1672       136   3673849.0       69.61
325    DISTRICT1  STATE33       4       534        98    957853.0       69.37
227  DISTRICT159  STATE28      18       870       134   2954367.0       66.23
86   DISTRICT667  STATE25       5       396        75    896129.0       82.23
425  DISTRICT144  STATE31       7      1064       108   2662077.0       71.59
260  DISTRICT305   STATE3       2        62         6    145538.0       69.88
281  DISTRICT385  STATE35       6       531        30    354972.0       75.00
262  DISTRICT552   STATE3       3       103         4    111997.0       52.23
253  DISTRICT168   STATE3       5       312        16    176385.0       82.14
301  DISTRICT551  STATE14       9       103        63    693281.0       88.29
356  DISTRICT494  STATE34      25      2179       223   3596292.0       70.95
165  DISTRICT196  STATE21      10      1354       119   1795092.0       77.52
565  DISTRICT308  STATE17       8       721       144    848868.0       86.54
388  DISTRICT281  STATE23       6       392        58    949159.0       73.92
461  DISTRICT619  STATE22       5       859        57   1064989.0       68.36
384  DISTRICT455  STATE23       9      1217        55   1063458.0       68.85
590   DISTRICT70  STATE20       7       427        84   1846993.0       80.30
343  DISTRICT354  STATE33       2       192        46   1260419.0       88.66
539  DISTRICT440  STATE17      15      1465       167   2887826.0       88.23
459  DISTRICT431  STATE22       9      1778       143   2363744.0       73.42
667  DISTRICT123  STATE11       3        80        16    237586.0       88.49
387  DISTRICT231  STATE23       6       657        63    530299.0       64.51
306   DISTRICT37   STATE4       7      1083        92    642923.0       68.38
213  DISTRICT347  STATE28      11       623        94   2228397.0       59.65
97    DISTRICT22   STATE2       7       182         7   2531583.0       87.12
78   DISTRICT247  STATE25       7       314        60   1332042.0       72.73
394  DISTRICT640  STATE24      17      1857       191   1802777.0       69.00
184  DISTRICT596  STATE21      11      1281       108   2149066.0       51.76
147  DISTRICT335  STATE21      17      1945       138   4380793.0       69.44
542  DISTRICT489  STATE17       7       749        63   1198810.0       85.14
105  DISTRICT157  STATE13      14      1994       508   3671999.0       71.68
254  DISTRICT458   STATE3       3       157        19     82839.0       76.33
109  DISTRICT158  STATE13       6       769       211   1338114.0       66.19
609   DISTRICT17  STATE20       4       359        59   9588910.0       88.48
53   DISTRICT126  STATE26       3       197        21    596294.0       68.90
81    DISTRICT45  STATE25       9       351       130   1742815.0       73.24
516  DISTRICT300   STATE9       5       651        84    590379.0       73.29
641  DISTRICT484   STATE6      15       333        83   1721179.0       74.92
650  DISTRICT145   STATE6      11       489       100   1614069.0       84.09
70    DISTRICT99  STATE25       4       279        43    558890.0       83.44
163  DISTRICT366  STATE21       9      1330        86   1579160.0       79.99

# Steps to construct a confidence interal

1. Identify a sample statistic
2. Choose a confidence level
3. Find the margin of error
4. Calculate the interval

In [13]:
sample_mean = sampled_data['OVERALL_LI'].mean()

In [15]:
sample_mean

74.22359999999999

In [17]:
estimated_standard_error = sampled_data['OVERALL_LI'].std() / np.sqrt(sampled_data.shape[0])

In [19]:
estimated_standard_error

1.4292043386558317

In [27]:
stats.norm.interval(confidence = 0.95, loc = sample_mean, scale = estimated_standard_error)

(71.42241096968617, 77.02478903031381)

you have a 95% confidence interval for the mean district literacy rate that stretches from about 71.4% to 77 %

95% CI: (71.42, 77.02)

The Department of Education will use your estimate of the mean district literavy rate to help make decisions about distributing finds to different states.

In [30]:
stats.norm.interval(confidence = 0.99, loc = sample_mean, scale = estimated_standard_error)

(70.54221358373107, 77.90498641626891)

You have a 99% confidence interval for the mean district literacy rate that stretches from about 70.5% to 77.9%

# Relationship between confidence level and confidence interval

As the confidence level gets higher, the confidence interval gets wider.

With a confidence level of 95%, the interval covers 5.6 percentage points (71.4% - 77.0%)
With a confidence level of 99%, the interval covers 7.4 percentage points (70.5% - 77.9%

This is because a wider confidence interval is more likely to include the actual population parameter)